plugins/ocm-backend/src/service/router.ts

```diff
@@ -45,30 +50,20 @@ import {
 } from '../helpers/parser';
 import { ManagedClusterInfo } from '../types';
 
-export interface RouterOptions {
-  logger: Logger;
-  config: Config;
-}
-
-export async function createRouter(
-  options: RouterOptions,
-): Promise<express.Router> {
-  const { logger } = options;
-  const { config } = options;
+const buildRouter = (config: Config, logger: Logger | LoggerService) => {
```

Comment:
We shouldn't be type unions here unless there is a really good reason to. This avoids complexity and prevents messy code going forward.